Parser/src/main/java/Grammar.java

```diff
@@ -0,0 +1,107 @@
+import lombok.Getter;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Getter
+public class Grammar {
+
+    private NonTerminal initialSymbol;
+    private List<NonTerminal> nonTerminals;
+    private List<Terminal> terminals;
+    private List<Production> productions = new ArrayList<>();
+
+    private Pair<List<String>, List<String>> splitProduction(String production) {
+        List<String> splittedProduction = List.of(production.split("->"));
+        return new Pair<>(Arrays.stream(splittedProduction.get(0).strip().split(" ")).collect(Collectors.toList()),
```

Comment:
Code readability could be improved here and in all the similar methods

Reply:
I definitely with Alex here. I'd make `Arrays.stream(string).strip().split("")).collect(Collectors.toList())` a separate function, while also checking the length before to be exactly 2 to have no surprises